components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,95 +461,138 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
+                    .isTokenLoggable(IdentityConstants.IdentityTokens.USER_ID_TOKEN)) {
+                log.debug("Retrieved the User Information:" + jwtAttributeMap);
+            }
 
-            processAuthenticatedUserScopes(context, oAuthResponse.getParam(OAuthConstants.OAuth20Params.SCOPE));
+            String authenticatedUserId = getAuthenticatedUserId(context, oAuthResponse, jwtAttributeMap);
+            String attributeSeparator = getMultiAttributeSeparator(context, authenticatedUserId);
 
-            AuthenticatedUser authenticatedUser;
-            Map<ClaimMapping, String> claims = new HashMap<>();
-            Map<String, Object> jsonObject = new HashMap<>();
+            jwtAttributeMap.entrySet().stream()
+                    .filter(entry -> !ArrayUtils.contains(NON_USER_ATTRIBUTES, entry.getKey()))
+                    .forEach(entry -> buildClaimMappings(claimsMap, entry, attributeSeparator));
 
-            if (StringUtils.isNotBlank(idToken)) {
-                jsonObject = getIdTokenClaims(context, idToken);
-                if (jsonObject == null) {
-                    String errorMessage = ErrorMessages.DECODED_JSON_OBJECT_IS_NULL.getMessage();
-                    if (log.isDebugEnabled()) {
-                        log.debug(errorMessage);
-                    }
-                    throw new AuthenticationFailedException(ErrorMessages.DECODED_JSON_OBJECT_IS_NULL.getCode(),
-                            errorMessage);
-                }
+            authenticatedUser = AuthenticatedUser
+                    .createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserId);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("The IdToken is null");
+            }
+            authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
+                    getAuthenticateUser(context, jwtAttributeMap, oAuthResponse));
+        }
+        claimsMap.putAll(getSubjectAttributes(oAuthResponse, authenticatorProperties));
+        authenticatedUser.setUserAttributes(claimsMap);
+        context.setSubject(authenticatedUser);
+    }
 
-                String idpName = context.getExternalIdP().getIdPName();
-                String sidClaim = (String) jsonObject.get(OIDCAuthenticatorConstants.Claim.SID);
-                if (StringUtils.isNotBlank(sidClaim) && StringUtils.isNotBlank(idpName)) {
-                    // Add 'sid' claim into authentication context, to be stored in the UserSessionStore for
-                    // single logout.
-                    context.setProperty(FEDERATED_IDP_SESSION_ID + idpName, sidClaim);
-                }
+    /**
+     * Retrieves or maps the ID token according to the flow supported by the authenticator.
+     * Overridden in Google Authenticator for Google one tap.
+     *
+     * @param context       AuthenticationContext.
+     * @param request       HttpServletRequest
```

Comment:
missing full stops.

Reply:
This is fixed 17aec3bd27b4fb46de26ee69631363ceabb32f2c